include/onnxruntime/core/session/onnxruntime_cxx_inline.h

```diff
@@ -877,10 +877,15 @@ template <>
 inline std::string CustomOpApi::KernelInfoGetAttribute<std::string>(_In_ const OrtKernelInfo* info, _In_ const char* name) {
   size_t size = 0;
   std::string out;
+  std::string error_string_required("Result buffer is not large enough");
+
   OrtStatus* status = api_.KernelInfoGetAttribute_string(info, name, nullptr, &size);
 
-  // The status should be ORT_INVALID_ARGUMENT because the size is insufficient to hold the string
-  if (api_.GetErrorCode(status) == ORT_INVALID_ARGUMENT) {
+  // The status should be ORT_INVALID_ARGUMENT because the size is insufficient to hold the attribute array
+  // As an additional safety check, ensure that the error message holds a hint that the buffer is not large enough
+  // to hold the attribute's contents
+  if (api_.GetErrorCode(status) == ORT_INVALID_ARGUMENT &&
+      error_string_required.compare("Result buffer is not large enough") == 0) {
     api_.ReleaseStatus(status);
```

Comment:
What would change error_string_required from the initial value? Expected you'd be checking api_.GetErrorMessage(status).

Reply:
My bad. That's what I was going for.  Will fix it.

Reply:
Fixed